lib/aot/compiler.cpp

```diff
@@ -2961,61 +2965,111 @@ class FunctionCompiler {
     const size_t ArgSize = FuncType.getParamTypes().size();
     const size_t RetSize =
         RTy->isVoidTy() ? 0 : FuncType.getReturnTypes().size();
-
-    llvm::Value *Args;
-    if (ArgSize == 0) {
-      Args = llvm::ConstantPointerNull::get(Builder.getInt8PtrTy());
-    } else {
-      auto *Alloca = Builder.CreateAlloca(Builder.getInt8Ty(),
-                                          Builder.getInt64(ArgSize * kValSize));
-      Alloca->setAlignment(Align(kValSize));
-      Args = Alloca;
+    std::vector<llvm::Value *> ArgsVec(ArgSize + 1, nullptr);
+    ArgsVec[0] = F->arg_begin();
+    for (size_t I = 0; I < ArgSize; ++I) {
+      const size_t J = ArgSize - 1 - I + 1;
```

Comment:
Not sure why you `-1` and then `+1` here.

Reply:
Removed.